crml/transaction-payment/src/lib.rs

```diff
@@ -310,9 +319,10 @@ where
 
 		// Only mess with balances if the fee is not zero.
 		if !fee.is_zero() {
+			let payer = T::FeePayer::fee_payer(call).unwrap_or(who.clone());
```

Comment:
```suggestion
			let payer = T::FeePayer::fee_payer(call).unwrap_or_else(|| who.clone());
```
might help with the clone

Reply:
Would you explain more?

Reply:
`unwrap_or` will eagerly evaluate the `.clone()`, `unwrap_or_else` will only evaluate it when needed (if payer is None)

Reply:
Ah you mean we do the clone only if necessary, right?

Reply:
But I'm not sure what would be stored for the closure then. It may defeat the purpose.